python/ray/serve/schema.py

```diff
@@ -487,6 +517,118 @@ def get_empty_schema_dict() -> Dict:
         return {"applications": []}
 
 
+@PublicAPI(stability="alpha")
+class DeploymentDetails(BaseModel, extra=Extra.forbid):
+    name: str = Field(description="Deployment name.")
+    deployment_status: DeploymentStatus = Field(
+        description="The current status of the deployment."
+    )
+    message: str = Field(
+        description=(
+            "If the deployment status is UNHEALTHY, this will describe the issue in "
+            "more detail."
+        )
+    )
+    deployment_config: DeploymentSchema = Field(
+        description=(
+            "The set of deployment config options that are currently applied to this "
+            "deployment. These options may come from users code, or from override "
+            "options specified in the config file applied to the cluster, or could be "
+            "default values loaded in if unspecified."
+        )
+    )
+
+    @validator("deployment_config")
+    def deployment_route_prefix_not_set(cls, v: DeploymentSchema):
+        # Route prefix should not be set at the deployment level. Deployment-level route
+        # prefix is outdated, there should be one route prefix per application
+        if "route_prefix" in v.dict(exclude_unset=True):
```

Comment:
Could you also explicitly check that `route_prefix` is not `None` here? If it's somehow explicitly set to `None`, it'll still show up even if `exclude_unset` is `True`.

Reply:
The problem is `None` can be a valid value for `route_prefix`, if the user doesn't want their ingress deployment/application to be exposed over HTTP. So I'm trying to ensure that the value will just never be set at the deployment level.

Reply:
Ah, that makes sense. Could you update the `ValueError` message to say `"Unexpectedly found a deployment-level route_prefix..."` instead of `"...non-null route_prefix..."`? Otherwise, when the user sets a deployment-level `None` route prefix, the error message would be confusing.

Reply:
Thanks for the suggestion, changed!